src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,160 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+
+    private RegistrationService registrationService;
+    private User user;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        Storage.people.clear();
+        user = new User();
+        user.setLogin("Example1");
+        user.setPassword("Password1");
+        user.setAge(30);
+        Storage.people.add(user);
+    }
+
+    @Test
+    void registerExistingLogin_NotOk() {
+        User newUser = new User();
+        newUser.setLogin("Example1");
+        newUser.setPassword("DifferentPass");
+        newUser.setAge(25);
+        assertThrows(IllegalArgumentException.class, () ->
+                registrationService.register(newUser));
+    }
+
+    @Test
+    void registerRepeatedLogin_Ok() {
+        User newUser = new User();
+        newUser.setLogin("PashaPasha");
+        newUser.setPassword("ProperPassword2");
+        newUser.setAge(28);
+        User registeredUser = registrationService.register(newUser);
+        assertNotNull(registeredUser);
+        assertEquals("PashaPasha", registeredUser.getLogin());
+        assertEquals("ProperPassword2", registeredUser.getPassword());
+        assertEquals(Integer.valueOf(28), registeredUser.getAge());
+    }
```

Comment:
The test name `registerRepeatedLogin_Ok` suggests that it's acceptable to register a user with a login that already exists, which contradicts the uniqueness requirement. Ensure your test names accurately reflect the expected behavior.